src/pip/_internal/configuration.py

```diff
@@ -362,12 +355,9 @@ def get_environ_vars(self):
         # type: () -> Iterable[Tuple[str, str]]
         """Returns a generator with all environmental vars with prefix PIP_"""
         for key, val in os.environ.items():
-            should_be_yielded = (
-                key.startswith("PIP_") and
-                key[4:].lower() not in self._ignore_env_names
-            )
-            if should_be_yielded:
-                yield key[4:].lower(), val
+            name = key[4:].lower()
+            if key.startswith("PIP_") and name not in ENV_NAMES_IGNORED:
+                yield name, val
```

Comment:
Nit, the previous implementation is likely slight more performant due to short-circuiting skips the `lower()` call. This can be significant since we need to read all the environment variables

Reply:
Thanks, I forgot that there's virtually no limit on how long the value an unrelated env var might be.